src/main/java/com/azure/cosmos/workedappexample/SampleGroceryStore.java

```diff
@@ -0,0 +1,326 @@
+                .subscribeOn(Schedulers.elastic())
+                .doOnSuccess(aVoid -> {
+                    isProcessorRunning.set(true);
+                })
+                .subscribe();
+
+            while (!isProcessorRunning.get()); //Wait for Change Feed processor start
+
+            System.out.println("\n\n\n\nPress enter to insert 10 items into the container." + COLLECTION_NAME + "...");
+            c.readLine();
+
+            // Insert 10 documents into the feed container
+            // createNewDocumentsJSON demonstrates how to insert a JSON object into a Cosmos DB container as an item
+            createNewDocumentsJSON(feedContainer, Duration.ofSeconds(3));
+
+            System.out.println("\n\n\n\nPress enter to delete item with id " + idToDelete + " (Jerry's plums)...");
+            c.readLine();
+
+            // Demonstrate deleting a document from the feed container and the materialized view, using TTL=0
+            deleteDocument();
+
+            System.out.println("\n\n\n\nPress ENTER to clean up & exit the sample code...");
+            c.readLine();
+
+            if (changeFeedProcessorInstance != null) {
+                changeFeedProcessorInstance.stop().block();
+            }
+
+            logger.info("-->DELETE sample's database: " + DATABASE_NAME);
+            deleteDatabase(cosmosDatabase);
+
+            Thread.sleep(500);
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        System.out.println("END Sample");
+    }
+
+    public static ChangeFeedProcessor getChangeFeedProcessor(String hostName, CosmosAsyncContainer feedContainer, CosmosAsyncContainer leaseContainer) {
+        ChangeFeedProcessorOptions cfOptions = new ChangeFeedProcessorOptions();
+        cfOptions.setFeedPollDelay(Duration.ofMillis(100));
+        cfOptions.setStartFromBeginning(true);
+        return ChangeFeedProcessor.changeFeedProcessorBuilder()
+            .setOptions(cfOptions)
+            .setHostName(hostName)
+            .setFeedContainer(feedContainer)
+            .setLeaseContainer(leaseContainer)
+            .setHandleChanges((List<JsonNode> docs) -> {
+                for (JsonNode document : docs) {
+                        //Duplicate each document update from the feed container into the materialized view container
+                        updateInventoryTypeMaterializedView(document);
+                }
+
+            })
+            .build();
+    }
+
+    private static void updateInventoryTypeMaterializedView(JsonNode document) {
+        typeContainer.upsertItem(document).subscribe();
+    }
+
+    public static CosmosAsyncClient getCosmosClient() {
+
+        return new CosmosClientBuilder()
+                .setEndpoint(AccountSettings.HOST)
+                .setKey(AccountSettings.MASTER_KEY)
+                .setConnectionPolicy(ConnectionPolicy.getDefaultPolicy())
+                .setConsistencyLevel(ConsistencyLevel.EVENTUAL)
```

Comment:
Eventual consistency level can lead to duplicate entries. Let's change this to SESSION instead.